### ISSUE TITLE
"Last updated" column dates

### ISSUE THREAD

**Issue by ConnorKrammer** [erengy/taiga#149]
I like the last updated column because it lets me know what order I last viewed shows in on my currently watching list, among other things. It's helpful how it displays recent dates in relative terms, such as "1 day ago" or "2 days ago" etc., but I noticed something odd when looking at it today. It seems that once the current month changes, the displayed text stops using relative dates and switches to absolute ones -- including shows that were updated just two or three days ago.

I'd suggest leaving the relative dates for updates that are less than two weeks old, even across month boundaries. Generally speaking, I'm more interested in how long ago a change was than the specific date it occurred on, especially for more recent changes. In an ideal world, there would be a small icon in the "Last Updated" column header that would allow toggling between all absolute/all relative/mixed dates (though I can understand why there isn't).

Edit: A reasonable compromise could be to display relative dates *on hover* when the user mouses over an absolute date. This isn't as easy to scan as if the entire column could be toggled, but still provides the information to people who want it without hurting those who don't.

**Comment by erengy** [comment]
I agree that the boundaries need to be modified in the way you suggest; they've been bothering me as well. Is there any standard in regards to this? If not, I may replicate the behavior of some popular JavaScript code (e.g. [time-elements](https://github.com/github/time-elements), [jquery-timeago](https://github.com/rmm5t/jquery-timeago)) to some extent.

I think it makes sense to display absolute dates after a certain point. I may push that point back to as far as 1.5 years ago. Do you think that'd be appropriate? For instance, if you last watched an anime *9 months ago*, would you prefer seeing that or *2014-11-03*?

I don't think I'll change the actual column text on hover, but I can make it so that they display a tooltip.

Notes to self:

- Consider displaying a tooltip for the Season column as well.
- While it should never happen, the relevant function needs to be able to handle future dates too.

**Comment by ConnorKrammer** [comment]
I'm not aware of any standards, unfortunately. I did a little looking around at places I know display relative dates. Reddit, for example, *always* displays the date in relative format, which makes sense given the site's frequently-updated content. GitHub itself displays dates in relative format under the "last commit" column of the repo page, and uses a mixture of relative/absolute on issue pages (cutoff boundary: 30 days).

I think a 30 day cutoff is pretty reasonable, and nicely coincides with a single month. I wouldn't make it longer than a year though, at maximum -- in fact, I don't think I'd make it much longer than 90 days. Anything too far in the past will be out of the user's memory, and at that point I think it's better to give the absolute date instead of the relative one, especially since the relative date loses resolution when it hits the "*# months ago*" stage. Also consider that if additional sort columns are implemented as in #150, with resolution of one month, the start, end, and last updated dates would quite often display the same values.

I do like the idea of having a toggle in the settings, as you suggested in #150. Especially if tooltips are shown (that's actually what I meant to say when I said "display relative dates on hover", but I wasn't very clear), that would be user-friendly but still detailed enough for the users that want it.

I like the idea of showing the exact air date in a tooltip in the Season column, as well.

**To summarize**: I propose the default be relative dates for the last 30 days, and absolute dates prior to this. A toggle setting would allow showing all relative dates (for all columns). Dates would, on hover, display the alternate time format in a tooltip (hover over absolute -> relative in tooltip, hover over relative -> absolute in tooltip).

**Comment by erengy** [comment]
5b08fef441f553aa0858aacd5c021f7eed8d58d3 and b7a4034255acd04ab9adafa71766d262b278c8fd handles this issue, for the most part. Currently the column always displays relative time. I'm planning to improve the tooltips so that "7 hours ago" has a tooltip text of "10:02:23" rather than "August 05". You can try out the [latest build](http://taiga.erengy.com/latest.html), if you want.

**Comment by ConnorKrammer** [comment]
I tried it out and I like the changes. I'd suggest leaving off the seconds on the "# hours ago" tooltip, though. The user probably doesn't care about the exact second, and it looks cleaner without it. You might also want to put "AM" or "PM" where appropriate.

Incidentally, I did get something odd on the progress bars (the grey line through the middle), and I'll open an issue for it if it isn't intentional.
![Progress Bar](https://i.imgur.com/6dNIpco.png)

**Comment by erengy** [comment]
The lines are just [something](http://myanimelist.net/forum/?topicid=214500&show=450#msg40956893) I've been experimenting with (it's not committed to the public repository). They display the available episodes on your library folders.

**Comment by ConnorKrammer** [comment]
Oh cool. That's actually something I've been thinking about myself. Would you like any thoughts I have on it?

**Comment by erengy** [comment]
Yeah, sure. Let's open a new issue for it though, because I'm planning to close this one tomorrow after another commit.

**Comment by ConnorKrammer** [comment]
I'm in the process of opening one.

**Comment by ConnorKrammer** [comment]
Just some feedback after using the latest build for a while now:

Using relative time measurements for all dates works really well. My arguments against it as the default still stand, particularly the resolution one, but that is largely fixed with the addition of tooltips. I still think there should be an option to toggle absolute/relative, but I'm leaning towards having relative be the new default. The only caveat with this is that none of my list items are older than a year, so I can't speak to if the resolution loss there is an issue.
